pkg/sbproxy/notifications/handlers/broker_handler.go

```diff
@@ -0,0 +1,196 @@
+	CatalogFetcher platform.CatalogFetcher
+
+	ProxyPrefix string
+	ProxyPath   string
+}
+
+// OnCreate creates brokers from the specified notification payload by invoking the proper platform clients
+func (bnh *BrokerResourceNotificationsHandler) OnCreate(ctx context.Context, payload json.RawMessage) {
+	brokerPayload := brokerPayload{}
+	if err := json.Unmarshal(payload, &brokerPayload); err != nil {
+		log.C(ctx).WithError(err).Error("error unmarshaling broker create notification payload")
+		return
+	}
+
+	if err := brokerPayload.Validate(notifications.CREATED); err != nil {
+		log.C(ctx).WithError(err).Error("error validating broker payload")
+		return
+	}
+
+	brokerToCreate := brokerPayload.New
+
+	existingBroker, err := bnh.BrokerClient.GetBrokerByName(ctx, bnh.ProxyPrefix+brokerToCreate.Resource.GetID())
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("error finding broker with name %s in the platform", brokerToCreate.Resource.GetID())
+		return
+	}
+
+	if existingBroker != nil && shouldBeProxified(existingBroker, brokerToCreate.Resource) {
+		updateRequest := &platform.UpdateServiceBrokerRequest{
+			GUID:      existingBroker.GUID,
+			Name:      bnh.ProxyPrefix + brokerToCreate.Resource.GetID(),
+			BrokerURL: bnh.ProxyPath + "/" + brokerToCreate.Resource.GetID(),
+		}
+		if _, err := bnh.BrokerClient.UpdateBroker(ctx, updateRequest); err != nil {
+			log.C(ctx).WithError(err).Errorf("error proxifying platform broker with GUID %s with SM broker with id %s", existingBroker.GUID, brokerToCreate.Resource.GetID())
+			return
+		}
+	} else {
+		createRequest := &platform.CreateServiceBrokerRequest{
+			Name:      bnh.ProxyPrefix + brokerToCreate.Resource.GetID(),
+			BrokerURL: bnh.ProxyPath + "/" + brokerToCreate.Resource.GetID(),
+		}
+		if _, err := bnh.BrokerClient.CreateBroker(ctx, createRequest); err != nil {
+			log.C(ctx).WithError(err).Errorf("error creating broker with name %s url %s", createRequest.Name, createRequest.BrokerURL)
+			return
+		}
+	}
+}
+
+// OnUpdate modifies brokers from the specified notification payload by invoking the proper platform clients
+func (bnh *BrokerResourceNotificationsHandler) OnUpdate(ctx context.Context, payload json.RawMessage) {
+	brokerPayload := brokerPayload{}
+
+	if err := json.Unmarshal(payload, &brokerPayload); err != nil {
+		log.C(ctx).WithError(err).Error("error unmarshaling broker create notification payload")
+		return
+	}
+
+	if err := brokerPayload.Validate(notifications.MODIFIED); err != nil {
+		log.C(ctx).WithError(err).Error("error validating broker payload")
+		return
+	}
+
+	brokerAfterUpdate := brokerPayload.New
+
+	existingBroker, err := bnh.BrokerClient.GetBrokerByName(ctx, bnh.ProxyPrefix+brokerAfterUpdate.Resource.GetID())
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("error finding broker with name %s in the platform", brokerAfterUpdate.Resource.GetID())
+		return
+	}
+	if existingBroker.BrokerURL != bnh.ProxyPath+"/"+brokerAfterUpdate.Resource.GetID() {
```

Comment:
minor: 
bnh.ProxyPath+"/"+brokerAfterUpdate.Resource.GetID() 
is calculated in different places (only with different variable names). Maybe it could be extracted as a function as the logic might change in time.